docs-rtd/reference/upgrading-provider.md

```diff
@@ -0,0 +1,235 @@
+# Upgrading to Juju Terraform Provider v1.0.0
+
+This guide outlines the breaking changes and upgrade procedures when migrating from pre-v1.0.0 versions of the Juju Terraform provider to v1.0.0.
+
+## Breaking Changes
+
+### 1. Model Field Replacement
+
+The most significant change is the replacement of the `model` field with `model_uuid` across multiple resources and data sources.
+
+#### Affected Resources:
+- `juju_application`
+- `juju_offer` 
+- `juju_ssh_key`
+- `juju_access_model`
+- `juju_access_secret`
+- `juju_integration`
+- `juju_secret`
+- `juju_machine`
+
+#### Affected Data Sources:
+- `juju_model`
+- `juju_application`
+- `juju_secret`
+- `juju_machine`
+
+**Before v1.0.0:**
+```terraform
+resource "juju_application" "app" {
+  name  = "postgresql"
+  model = juju_model.test.name
+  charm {
+    name = "postgresql"
+  }
+}
+
+data "juju_model" "existing" {
+  name = "my-model"
+}
+```
+
+**After v1.0.0:**
+```terraform
+resource "juju_application" "app" {
+  name       = "postgresql"
+  model_uuid = juju_model.test.uuid # <-- 
+  charm {
+    name = "postgresql"
+  }
+}
+
+data "juju_model" "existing" {
+  uuid = "model-uuid-here" # <--
+}
+```
+
+This change does not require resource re-creation. All infrastructure should remain intact
+and only the Terraform state will be updated.
+
+### 2. Import Syntax Changes
+
+Import syntax for model-scoped resources has changed to require model UUIDs instead of model names.
+
+**Before v1.0.0:**
+```bash
+terraform import juju_application.myapp model-name:application-name
+```
+
+**After v1.0.0:**
+```bash
+terraform import juju_application.myapp model-uuid:application-name
+```
+
+### 3. Offer Data Source Changes
+
+The `juju_offer` data source no longer contains the computed `model` field.
+
+### 4. Application Resource Field Removals
+
+Several deprecated fields have been removed from the `juju_application` resource:
+
+- **`placement`** - Use `machines` instead. This is epected to cause resource replacement.
+- **`principle`** - Field was unused and has been removed.
+- **`series`** - Use `base` instead.
+
+### 5. Machine Resource Field Removals
+
+The deprecated `series` field has been removed from the `juju_machine` resource. Use `base` instead.
+
+**Before v1.0.0:**
+```terraform
+resource "juju_machine" "machine" {
+  model  = juju_model.test.name
+  series = "focal"
+}
+```
+
+**After v1.0.0:**
+```terraform
+resource "juju_machine" "machine" {
+  model_uuid = juju_model.test.uuid
+  base       = "ubuntu@20.04"
+}
+```
+
+## Automated Upgrade Tool
+
+The team provides an automated upgrade tool called `tf-upgrader` to help migrate your Terraform configurations.
```

Comment:
We should publish reference about this tool and also a how-to with the contents we have here below.